packages/sign-client/src/controllers/engine.ts

```diff
@@ -1283,36 +1301,42 @@ export class Engine extends IEngine {
   // ---------- Relay Events Router ----------------------------------- //
 
   private registerRelayerEvents() {
-    this.client.core.relayer.on(
-      RELAYER_EVENTS.message,
-      async (event: RelayerTypes.MessageEvent) => {
-        const { topic, message } = event;
-
-        // Retrieve the public key (if defined) to decrypt possible `auth_request` response
-        const { publicKey } = this.client.auth.authKeys.keys.includes(AUTH_PUBLIC_KEY_NAME)
-          ? this.client.auth.authKeys.get(AUTH_PUBLIC_KEY_NAME)
-          : ({ responseTopic: undefined, publicKey: undefined } as any);
-
-        const payload = await this.client.core.crypto.decode(topic, message, {
-          receiverPublicKey: publicKey,
-        });
+    this.client.core.relayer.on(RELAYER_EVENTS.message, (event: RelayerTypes.MessageEvent) => {
+      // capture any messages that arrive before the client is initialized so we can process them after initialization is complete
+      if (!this.initialized || this.relayMessageCache.length > 0) {
+        this.relayMessageCache.push(event);
+      } else {
+        this.onRelayMessage(event);
+      }
+    });
+  }
 
-        try {
-          if (isJsonRpcRequest(payload)) {
-            this.client.core.history.set(topic, payload);
-            this.onRelayEventRequest({ topic, payload });
-          } else if (isJsonRpcResponse(payload)) {
-            await this.client.core.history.resolve(payload);
-            await this.onRelayEventResponse({ topic, payload });
-            this.client.core.history.delete(topic, payload.id);
-          } else {
-            this.onRelayEventUnknownPayload({ topic, payload });
-          }
-        } catch (error) {
-          this.client.logger.error(error);
-        }
-      },
-    );
+  private async onRelayMessage(event: RelayerTypes.MessageEvent) {
+    const { topic, message } = event;
+
+    // Retrieve the public key (if defined) to decrypt possible `auth_request` response
+    const { publicKey } = this.client.auth.authKeys.keys.includes(AUTH_PUBLIC_KEY_NAME)
+      ? this.client.auth.authKeys.get(AUTH_PUBLIC_KEY_NAME)
+      : ({ responseTopic: undefined, publicKey: undefined } as any);
```

Comment:
```suggestion
    const publicKey = this.client.auth.authKeys.keys.includes(AUTH_PUBLIC_KEY_NAME)
      ? this.client.auth.authKeys.get(AUTH_PUBLIC_KEY_NAME).publicKey
      : undefined;
```

Reply:
it has been this way for a long time so would avoid changing it in this PR, it was just copy/paste to a new line